src/state/router/selectors.js

```diff
@@ -59,39 +60,41 @@ export function getUrlParam(state, key, defaultsTo) {
   return defaultsTo;
 }
 
-export function getUrlParamAsInt(state, name, defaultsTo) {
+export function getUrlParamAsInt(state, name, defaultsTo = null) {
   return parseInt(getUrlParam(state, name, defaultsTo), 10);
 }
 
-export function getAllQueryParams(state, defaultsTo) {
-  const search = state.router.location.search;
+export function getAllQueryParams(state, defaultsTo = null) {
+  const search = state.getIn(['router', 'location', 'search']);
 
   if (!search) {
     return defaultsTo;
   }
 
-  return search
-    .slice(1)
-    .split('&')
-    .map(item => item.split('='))
-    .reduce((obj, [key, value]) => {
-      obj[key] = value === undefined ? true : value;
-      return obj;
-    }, {});
+  return fromJS(
```

Comment:
I'm confused, isn't `search` taken from the state, which is fully immutable? Why the `fromJS`?
Also it feels weird to have a `fromJS` here, and in `getCurrentUrl` do a `toJS` on the returned value from here...

Reply:
`search` is taken from the state. It is however a string so it's not immutable. We subsequently split it (transforming it to a standard `Array`) and perform some additional processing on it. The `fromJS` here converts the standard `Array` into an immutable `List`.

Reply:
Perhaps i should add a comment explaining

Reply:
ah, makes sense, thanks for the explanation. And yeah, would love a comment ;)